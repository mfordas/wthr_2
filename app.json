{
  "name": "wthr_2",
  "displayName": "wthr_2"
}