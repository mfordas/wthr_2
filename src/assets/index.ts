import Humidity from './images/humidity.png';
import Night from './images/night.png';
import Pin from './images/pin.png';
import Sun from './images/sun.png';
import Weather from './images/weather.png';

export const Images = {
  Pin,
  Humidity,
  Night,
  Weather,
  Sun,
};
