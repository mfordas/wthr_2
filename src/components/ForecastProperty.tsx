import {round} from 'mathjs';
import React from 'react';
import {Image, ImageSourcePropType, StyleSheet, Text, View} from 'react-native';

interface Props {
  imgSrc?: ImageSourcePropType;
  value: string | number;
  unit?: string;
}

export const ForecastProperty: React.FC<Props> = ({imgSrc, value, unit}) => {
  return (
    <View style={styles.item}>
      {imgSrc && <Image source={imgSrc} style={styles.image} />}
      <Text>
        {typeof value === 'number' ? round(value) : value}
        {unit}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  item: {
    width: '20%',
    height: 68,
    justifyContent: 'center',
    alignItems: 'center',
  },
  image: {
    width: 24,
    height: 24,
    margin: 12,
  },
});
