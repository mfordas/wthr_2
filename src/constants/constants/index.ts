import Config from 'react-native-config';

export const CELSIUS_DEG = '\u00b0C';
export const HUMIDITY_UNIT = '%';

export const API_KEY = Config.API_KEY;
