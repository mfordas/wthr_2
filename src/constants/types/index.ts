export interface apiGeocodeCityResponse {
  name: string;
  local_names: Record<string, string>;
  lat: number;
  lon: number;
  country: string;
  state?: string;
}

export type apiGeocodeResponse = apiGeocodeCityResponse[];

export interface apiCityResponse {
  id: number;
  name: string;
  coord: {
    lat: number;
    lon: number;
  };
  country: string;
  population: number;
  timezone: number;
  sunrise: number;
  sunset: number;
}

export interface apiOneDayForecastResponse {
  dt: number;
  sunrise: number;
  sunset: number;
  moonrise: number;
  moonset: number;
  moon_phase: number;
  temp: {
    day: number;
    min: number;
    max: number;
    night: number;
    eve: number;
    morn: number;
  };
  feels_like: {
    day: number;
    night: number;
    eve: number;
    morn: number;
  };
  pressure: number;
  humidity: number;
  dew_point: number;
  wind_speed: number;
  wind_deg: number;
  weather: [
    {
      id: number;
      main: string;
      description: string;
      icon: string;
    },
  ];
  clouds: number;
  pop: number;
  rain: number;
  uvi: number;
}

export interface apiFiveDaysForecastResponse {
  lat: number;
  lon: number;
  timezone: string;
  timezone_offset: number;
  daily: apiOneDayForecastResponse[];
}
