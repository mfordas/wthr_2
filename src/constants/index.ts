import * as constants from './constants';
import * as types from './types';

export {types, constants};
