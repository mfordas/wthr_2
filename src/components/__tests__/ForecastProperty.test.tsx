import {render} from '@testing-library/react-native';
import React from 'react';

import {Images} from 'app/assets';
import {constants} from 'app/constants';

import {ForecastProperty, Props} from '../ForecastProperty';

const props: Props = {
  imgSrc: Images.Humidity,
  value: 18,
  unit: constants.CELSIUS_DEG,
};
const wrapper = render(<ForecastProperty {...props} />);
test('ForecastProperty should match snapshot', () => {
  expect(wrapper).toMatchSnapshot();
});
