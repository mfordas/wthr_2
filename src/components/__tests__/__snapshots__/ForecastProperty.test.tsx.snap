// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastProperty should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": 68,
      "justifyContent": "center",
      "width": "20%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/humidity.png",
      }
    }
    style={
      Object {
        "height": 24,
        "margin": 12,
        "width": 24,
      }
    }
  />
  <Text>
    18
    Â°C
  </Text>
</View>
`;
